src/app/pages/psbt-request/use-psbt-request.tsx

```diff
@@ -1,68 +1,83 @@
 import { useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import { bytesToHex, hexToBytes } from '@noble/hashes/utils';
+import * as btc from '@scure/btc-signer';
 
 import { finalizePsbt } from '@shared/actions/finalize-psbt';
-import { logger } from '@shared/logger';
-import { ensureArray, isUndefined } from '@shared/utils';
+import { RouteUrls } from '@shared/route-urls';
+import { ensureArray, undefinedIfLengthZero } from '@shared/utils';
 
 import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { getPsbtPayloadFromToken } from '@app/common/psbt/requests';
+import { usePsbtRequestSearchParams } from '@app/common/psbt/use-psbt-request-params';
 import { usePsbtSigner } from '@app/features/psbt-signer/hooks/use-psbt-signer';
-import { usePsbtRequestSearchParams } from '@app/pages/psbt-request/psbt-request.hooks';
 
 export function usePsbtRequest() {
-  const { requestToken, tabId } = usePsbtRequestSearchParams();
   const [isLoading, setIsLoading] = useState(false);
-  const { signPsbt, signPsbtAtIndex, getDecodedPsbt, getPsbtAsTransaction } = usePsbtSigner();
+  const navigate = useNavigate();
+  const { origin, requestToken, tabId } = usePsbtRequestSearchParams();
+
+  const { signPsbt, getDecodedPsbt, getPsbtAsTransaction } = usePsbtSigner();
+
   const analytics = useAnalytics();
+
   return useMemo(() => {
     if (!requestToken) throw new Error('Cannot decode psbt without request token');
+
     const payload = getPsbtPayloadFromToken(requestToken);
     const payloadTxBytes = hexToBytes(payload.hex);
     const appName = payload?.appDetails?.name;
+    const allowedSighash = undefinedIfLengthZero(
+      payload?.allowedSighash?.map(h => Number(h)) as btc.SignatureHash[]
+    );
+    const indexesToSign = undefinedIfLengthZero(
+      ensureArray(payload?.signAtIndex)?.map(h => Number(h))
+    );
```

Comment:
Isn't this repeating logic that takes place in the request params hooks?

Reply:
I think what you are referencing is the rpc request params? The legacy ones were always handled diff in all the legacy requests. I can look at moving them though.

Reply:
Yeah, looked, I'll just move the legacy params into the hook like the rpc request does so handled the same. Just have to decode the `requestToken` there.